packages/core/src/validation/dispatch/helpers.ts

```diff
@@ -68,9 +89,33 @@ export async function validateBytesForArtifact(
     return null;
   }
 
+  return buildValidationError(
+    vertex,
+    `Artifact with name '${bytesArgs[bytesDoesNotExistIndex].label}' doesn't exist`,
+    callPoints
+  );
+}
+
+export function buildValidationError(
+  vertex:
+    | ArtifactContractDeploymentVertex
+    | HardhatContractDeploymentVertex
+    | EventVertex
+    | CallDeploymentVertex
+    | ArtifactLibraryDeploymentVertex
+    | HardhatLibraryDeploymentVertex
+    | DeployedContractDeploymentVertex
+    | SendVertex,
```

Comment:
Can we define a `ValidationVertex` or similar that is this union and use that here instead

Reply:
Good point. I have switched out both lists. One is really just a any deployment graph vertex, the other is that with a constraint that it has args as well.